danuluma/dannstore
# #161370002 add POST and GET products' endpoints

#### What does this PR do?
Add products endpoints

#### Description of Task to be completed?
Add POST products endpoint
Add an endpoint to get all products
Add endpoint to edit productss
Add endpoint to delete products

#### How should this be manually tested?
* Setup the project as recommended in the setup guide
* Use postman to send a POST request to http://localhost:5000/dann/api/v2/products
* The POST request should have just the book id in the body with the format of  
```
         {
            "title": "test_book",
            "description": "An awesome read",
            "price": 100,
            "quantity": 5,
            "minimum": 4,
            "image_url": "url",
            "created_by": 0
            }
```
* You may now send a GET request to http://localhost:5000/dann/api/v2/products
* Take note of the generated product id
* Now send GET request to http://localhost:5000/dann/api/v2/products/<int:productID>
* You should get details of the specific product
* You may update the product by sending PUT request to http://localhost:5000/dann/api/v2/products/<int:productID> with a body of the format 
```
         {
            "title": "test_book",
            "description": "An awesome read",
            "price": 100,
            "quantity": 5,
            "minimum": 4,
            "image_url": "url"
    
            }
```

* You can delete the product by simply sending a blank DELETE request to the endpoint 
http://localhost:5000/dann/api/v2/products/<int:productID>

#### What are the relevant pivotal tracker stories?
[#161370002](https://www.pivotaltracker.com/story/show/161370002)
